### PR TITLE
Recipe with dependencies fix

### PR DESCRIPTION
**Pull request by richtja**
This commit moves dependency serialization to Dependency class from safeloader to runnable. This will enable the use of dependencies in recipe files.

Reference: #5918

### CODE REVIEW

File: selftests/functional/serial/requirements.py [hunk 1]
```diff
@@ -294,3 +302,28 @@ def test_job_dependency(self):
...
                     result.stdout_text,
                 )
+
+    @skipUnless(os.getenv("CI"), skip_install_message)
+    def test_dependency_recipe(self):
+        with script.Script(
+            os.path.join(self.tmpdir.name, "test_multiple_success.py"),
+            TEST_WITHOUT_DEPENDENCY,
+        ) as test:
+            dependency_recipe_data = DEPENDENCY_RECIPE.replace(
```
Comment (clebergnu) [inline selftests/functional/serial/requirements.py:312]: This works, but, I wonder why you didn't go with standard Python formatting strings here.

**Review by clebergnu — APPROVED** [review]
Hi @richtja, just have a question/style suggestion, but overall, LGTM. Thanks!

File: selftests/functional/serial/requirements.py (hunk 1 above)
Comment (richtja) [reply selftests/functional/serial/requirements.py:312]: I am not sure what do you mean, can you please give me an example? :sweat_smile:

Comment (clebergnu) [reply selftests/functional/serial/requirements.py:312]: I mean:

```python
DEPENDENCY_RECIPE_FMT = """
{{
  "kind": "avocado-instrumented",
  "uri": "{path}",
  "kwargs": {{"dependencies": [{{"type": "package", "name": "hello"}}]}}
}}
"""
```

And then:

```python
DEPENDENCY_RECIPE_FMT.format(path=path)
```

It's just a bit more Pythonic and predictable IMO.

Comment (richtja) [reply selftests/functional/serial/requirements.py:312]: Oh, right. This is definitely much better. I will send an update.

**Pull request merged**